docs/source/point-evaluation.rst

```diff
@@ -104,6 +106,302 @@ in parallel. There is no special API, but there are some restrictions:
 * Each process will get the same values.
 
 
+Firedrake API 2: Interpolation onto a vertex-only mesh
+------------------------------------------------------
+
+Firedrake's second API for evaluating functions at arbitrary points,
+interpolation onto a :func:`~.VertexOnlyMesh`, is designed for evaluating a
+function at many points and for creating expressions which contain point
+evaluations. It has been designed from the ground up to be entirely parallel
+compatible. Whilst :meth:`~.Function.at` produces a list of values, a
+cross-mesh interpolation onto :func:`~.VertexOnlyMesh` gives Firedrake
+:py:class:`~.Function`\s.
+
+This is discussed in detail in :cite:`nixonhill2023consistent` but, briefly,
+the idea is that the :func:`~.VertexOnlyMesh` is a mesh whose wich represents a
+point cloud domain. Each cell of the mesh is a vertex at a chosen location in
+space. As usual for a mesh, we represent values by creating functions in
+function spaces on it. The only function space that makes sense for a mesh
+whose cells are vertices is the space of piecewise constant functions, also
+known as the Polynomial degree 0 Discontinuous Galerkin (P0DG) space.
+
+Our vertex-only meshes are immersed in some 'parent' mesh. We perform point
+evaluation of a function :math:`f` defined in a function space
+:math:`V` on the parent mesh by interpolating into the P0DG space on the
+:func:`~.VertexOnlyMesh`. For example:
+
+.. code-block:: python3
+
```

Comment:
Import the lines of code from a test: https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-literalinclude